src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerRequestForInformationSole.java

```diff
@@ -0,0 +1,119 @@
+import static uk.gov.hmcts.divorce.divorcecase.model.access.Permissions.CREATE_READ_UPDATE;
+
+@Slf4j
+@RequiredArgsConstructor
+@Component
+public class CaseworkerRequestForInformationSole implements CCDConfig<CaseData, State, UserRole> {
+
+    public static final String CASEWORKER_REQUEST_FOR_INFORMATION_SOLE = "caseworker-request-for-information-sole";
+
+    public static final String REQUEST_FOR_INFORMATION_NOTIFICATION_FAILED_ERROR
+        = "Unable to send Request for Information Notification for Case Id: ";
+
+    private final CaseworkerRequestForInformationHelper helper;
+
+    private final RequestForInformationNotification requestForInformationNotification;
+
+    private final NotificationDispatcher notificationDispatcher;
+
+    @Override
+    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        new PageBuilder(configBuilder
+            .event(CASEWORKER_REQUEST_FOR_INFORMATION_SOLE)
+            .forAllStates()
+            .name("Request For Information")
+            .description("Request for information")
+            .showCondition("applicationType=\"soleApplication\"")
+            .showSummary()
+            .showEventNotes()
+            .endButtonLabel("Submit")
+            .aboutToSubmitCallback(this::aboutToSubmit)
+            .grant(CREATE_READ_UPDATE, CASE_WORKER)
+            .grantHistoryOnly(SUPER_USER, LEGAL_ADVISOR, JUDGE, SOLICITOR, CITIZEN, JUDGE))
+            .page("requestForInformation", this::midEvent)
+            .pageLabel("Request For Information")
+            .complex(CaseData::getRequestForInformationList)
+                .complex(RequestForInformationList::getRequestForInformation)
+                    .mandatory(RequestForInformation::getRequestForInformationSoleParties)
+                    .mandatory(RequestForInformation::getRequestForInformationName, "requestForInformationSoleParties=\"other\"")
+                    .mandatory(RequestForInformation::getRequestForInformationEmailAddress, "requestForInformationSoleParties=\"other\"")
+                    .mandatory(RequestForInformation::getRequestForInformationDetails)
+                .done()
+            .done();
+    }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> midEvent(CaseDetails<CaseData, State> details,
+                                                                  CaseDetails<CaseData, State> detailsBefore) {
+
+        log.info("{} midEvent callback invoked for Case Id: {}", CASEWORKER_REQUEST_FOR_INFORMATION_SOLE, details.getId());
+
+        List<String> errors = helper.areEmailsValid(details.getData());
+        if (!errors.isEmpty()) {
+            return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+                .errors(errors)
+                .build();
+        }
+
+        return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+            .data(details.getData())
+            .build();
+    }
```

Comment:
I think the notification client provides a useful error when information is invalid? Maybe we could reuse that/let it propagate from the aboutToSubmit callback rather than validating emails here and including validation logic in the helper

Reply:
Possibly?  This is essentially the same logic used for general emails.